test/integration/output_test.py

```diff
@@ -2,82 +2,90 @@
-        ("gens_eia860", "gen_eia923", 12 / 1, {}),
-        ("gens_eia860", "gen_fuel_by_generator_energy_source_eia923", 12 / 1, {}),
-        ("gens_eia860", "gen_fuel_by_generator_eia923", 12 / 1, {}),
-        ("gens_eia860", "gf_eia923", 12 / 1, {}),
-        ("gens_eia860", "hr_by_unit", 12 / 1, {}),
-        ("gens_eia860", "hr_by_gen", 12 / 1, {}),
-        ("gens_eia860", "fuel_cost", 12 / 1, {}),
-        ("gens_eia860", "capacity_factor", 12 / 1, {}),
-        pytest.param(
-            "gens_eia860",
-            "mcoe_generators",
-            12 / 1,
-            {"all_gens": False},
-            marks=pytest.mark.xfail(reason="MCOE has time coverage issues."),
-        ),
+        ("core_eia860__assn_boiler_generator", 1 / 1),
+        ("_out_eia__yearly_generators", 1 / 1),
+        ("out_eia860__yearly_ownership", 1 / 1),
+        ("out_eia__yearly_plants", 1 / 1),
+        ("out_eia__yearly_boilers", 1 / 1),
+        ("_out_eia__plants_utilities", 1 / 1),
+        ("out_eia__yearly_utilities", 1 / 1),
+        ("out_eia923__monthly_boiler_fuel", 12 / 1),
+        ("out_eia923__monthly_fuel_receipts_costs", 12 / 1),
+        ("out_eia923__monthly_generation", 12 / 1),
+        ("out_eia923__monthly_generation_fuel_by_generator_energy_source", 12 / 1),
+        ("out_eia923__monthly_generation_fuel_by_generator", 12 / 1),
+        ("out_eia923__monthly_generation_fuel_combined", 12 / 1),
+        ("_out_eia__monthly_heat_rate_by_unit", 12 / 1),
+        ("_out_eia__monthly_heat_rate_by_generator", 12 / 1),
+        ("_out_eia__monthly_fuel_cost_by_generator", 12 / 1),
+        ("_out_eia__monthly_capacity_factor_by_generator", 12 / 1),
     ],
 )
-def test_eia_outputs(fast_out, df1_name, df2_name, mult, kwargs):
+def test_eia_outputs(
+    test_table: str,
+    mult: int,
+    pudl_engine: sa.Engine,  # Required to ensure that the data is available.
+):
     """Check EIA output functions and date frequencies of output dataframes."""
-    pytest.skip(reason="Memory intensive, GHA CI failing. Migrate to dbt ASAP.")
-    df1 = fast_out.__getattribute__(df1_name)()
-    logger.info(f"Running fast_out.{df2_name}() with freq={fast_out.freq}.")
-    df2 = fast_out.__getattribute__(df2_name)(**kwargs)
-    logger.info(f"Found {len(df2)} rows in {df2_name}")
-    logger.info(f"Checking {df2_name} date frequency relative to {df1_name}.")
-    pv.check_date_freq(df1, df2, mult)
+    yearly_gens_name = "_out_eia__yearly_generators"
+
+    # Get columns needed for date frequency checking
+    # Include data_maturity if it exists in schema for proper incremental_ytd filtering
+    def get_date_freq_columns(table_name: str) -> list[str]:
+        """Get columns needed for date frequency checking."""
+        columns = ["report_date"]
+        resource = Resource.from_id(table_name)
+        if "data_maturity" in resource.get_field_names():
+            columns.append("data_maturity")
+        return columns
```

Comment:
This lets us read just 1-2 columns from the Parquet files, rather than the whole tables, which dramatically reduces memory usage and read times, since the frequency checker just needs the date index and data maturity columns. I guess I could have done the filtering here too, rather than over in check_date_freq.